### ISSUE TITLE
Getting Started

### ISSUE THREAD

**Issue by msshaylaj** [printdotio/printio-api#33]
I am interested in simply showing print-ready products on my site seamlessly (to allow customized product pages and cart--without leaving my site). The documentation isn't clear as to a step-by-step process for getting this done. I am a PHP developer with JSON knowledge, but need to know the best way to get this up on my Wordpress site.

**Comment by AustenB** [comment]
Do you plan on using a shopping cart plugin for Wordpress like Woo
Commerce?

Today we don't support print ready products with our web javascript widget.

On Tue, Aug 4, 2015 at 1:44 PM, msshaylaj notifications@github.com wrote:

> I am interested in simply showing print-ready products on my site
> seamlessly (to allow customized product pages and cart--without leaving my
> site). The documentation isn't clear as to a step-by-step process for
> getting this done. I am a PHP developer with JSON knowledge, but need to
> know the best way to get this up on my Wordpress site.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/printdotio/printio-api/issues/33.

## 

Austen Bernstein
Founder & CEO
printio

**Comment by msshaylaj** [comment]
Yes. I am currently using Woo Commerce. Not sure if I am using the proper term. What is a "print ready" product? I just want the customer to order one of my designs on a mug for example without leaving my site.

**Comment by AustenB** [comment]
We haven't yet started our woo commerce plugin. You would have to integrate
the WooCommerce Plugin with our API.

Is this something you are interested in doing? We might be able to provide
a Guide or some step by step info for it.

Will let tech team confirm now that we understand your goals a little
better.

On Tue, Aug 4, 2015 at 1:52 PM, msshaylaj notifications@github.com wrote:

> Yes. I am currently using Woo Commerce. Not sure if I am using the proper
> term. What is a "print ready" product? I just want the customer to select
> one of my designs on a mug for example.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/printdotio/printio-api/issues/33#issuecomment-127691509
> .

## 

Austen Bernstein
Founder & CEO
printio

**Comment by msshaylaj** [comment]
Yes, a Woo Commerce integration with your API would be great. Let me know if there will be an extra cost involved to make that happen. If so, I can work with having a separate area of my site with your API that I can skin.

**Comment by AustenB** [comment]
Are you saying you will code an integration or you want to wait until we
have the plugin for you?

Just Confirming my understanding.

On Tue, Aug 4, 2015 at 1:56 PM, msshaylaj notifications@github.com wrote:

> Yes, a Woo Commerce integration with your API would be great. Let me know
> if there will be an extra cost involved to make that happen. If so, I can
> work with having a separate area of my site with your API that I can skin.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/printdotio/printio-api/issues/33#issuecomment-127692400
> .

## 

Austen Bernstein
Founder & CEO
printio

**Comment by msshaylaj** [comment]
I will code an integration. Just need some guidance as to how to get it up.

**Comment by micahasmith** [comment]
I would recommend going to our staging documentation site (https://staging-api.print.io/docs/) and:
1. placing your recipeid in the top right textbox and hitting "explore"
2. clicking into the /preconfiguredproducts endpoint and doing the GET

That endpoint contains the data for your "print ready" products (PRPs). Have you set up any PRPs in the admin site?

**Comment by msshaylaj** [comment]
Just attempted to create a "print ready" product for the sake of example and got this error:
Error. Must consist of only one item (currently).

**Comment by micahasmith** [comment]
Send me your json, pls?

**Comment by msshaylaj** [comment]
I haven't gotten to that point. I was just wondering where in the world to start in order to get this integrated with my Wordpress site. Sounds like I have to set up a PRP first correct?

**Comment by micahasmith** [comment]
Yep. So here's the flow:
1. Create a PRP. You _could_ do this via the API, but our admin site makes it super easy as well. (Especially if you're just testing)
2. Integrate the /PRP GET call into your site. Since by this point you'll have data in our db, the endpoint will return all the info you need to list out your PRPs.
3. Order submission. 

If you need help getting 1 accomplished, feel free to email me at micah at print dot io and i'll connect you with one of our partner support specialists to walk through the flow/details.

**Comment by msshaylaj** [comment]
Just tried to again to set up a PRP in the admin site and it keeps giving an error:
"Print Ready Product Creation Fails! Error. Must consist of only one item (currently)."

**Comment by micahasmith** [comment]
Are you trying to sell a product that consists of two items? Right now we only support single item PRPs.

**Comment by msshaylaj** [comment]
Nope. It's just a single pillow.

**Comment by micahasmith** [comment]
Ah! Ok. So the pillow requires two images, which is a feature we will be supporting in a week.

If you were attempting to create a canvas or a phone case, those would work (since there is no front/back to a canvas/phone case).

**Comment by msshaylaj** [comment]
I see. So I will just stand by until that's working I suppose. I will look for the announcement!

**Comment by micahasmith** [comment]
@msshaylaj it's ready! feel free to submit multi-image PRPs

**Comment by micahasmith** [comment]
@msshaylaj need any further help with getting up and going?

**Comment by msshaylaj** [comment]
Hi there,
Thanks for the follow up. I am just getting back from vacation and back on
the grind. I will be working on it again tomorrow and be sure to let you
know!
